browser/ui/tabs/shared_pinned_tab_service.cc

```diff
@@ -196,6 +200,13 @@ void SharedPinnedTabService::OnBrowserAdded(Browser* browser) {
 }
 
 void SharedPinnedTabService::OnBrowserSetLastActive(Browser* browser) {
+  if (change_source_model_) {
```

Comment:
2. An then I ran into this. attaching a restored web contents seems to activate the target browser, which causes reentrancy.